src/Microsoft.OData.Core/Json/ODataJsonWriterUtils.cs

```diff
@@ -224,27 +224,40 @@ private static void WriteInnerError(IJsonWriter jsonWriter, ODataInnerError inne
             jsonWriter.WriteName(innerErrorPropertyName);
             jsonWriter.StartObjectScope();
 
-            //// NOTE: we add empty elements if no information is provided for the message, error type and stack trace
-            ////       to stay compatible with Astoria.
+            if (innerError.Properties != null)
+            {
+                foreach (KeyValuePair<string, string> pair in innerError.Properties)
+                {
+                    jsonWriter.WriteName(pair.Key);
+                    jsonWriter.WriteValue(pair.Value);
+                }
+            }
+            else

```

Comment:
>     else [](start = 7, length = 9)

Should this really be else? does that mean we either write the open properties or the defined properties, but not both?